game/src/contrib/systems/CollisionSystem.java

```diff
@@ -12,16 +12,15 @@
 /** System to check for collisions between two entities */
 public class CollisionSystem extends System {
 
-    private record CollisionKey(int a, int b) {}
-
-    protected record CollisionData(CollideComponent a, CollideComponent b) {}
+    private final Map<CollisionKey, CollisionData> collisions = new HashMap<>();
 
-    private Map<CollisionKey, CollisionData> collisions = new HashMap<>();
+    public CollisionSystem() {
+        super(CollideComponent.class);
+    }
 
-    /** checks if there is a collision between two entities based on their hitbox */
     @Override
-    public void update() {
-        Game.getEntities().stream()
+    public void execute() {
+        getEntityStream()
```

Comment:
Auweia :)  Der Code hier drin gehört in mind. drei Teilmethoden aufgesplittet, so wie es aussieht.

Reply:
das ding pack ich nicht an.

Reply:
#682